app/Http/Controllers/Api/ApiPetController.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Pet;
+use Validator;
+use App\Contact;
+use Illuminate\Http\Request;
+use Illuminate\Database\QueryException;
+use App\Http\Resources\Pet\Pet as PetResource;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class ApiPetController extends ApiController
+{
+    /**
+     * Get the detail of a given pet.
+     * @param  Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Request $request, $id)
+    {
+        try {
+            $pet = Pet::where('account_id', auth()->user()->account_id)
+                ->where('id', $id)
+                ->firstOrFail();
+        } catch (ModelNotFoundException $e) {
+            return $this->respondNotFound();
+        }
+
+        return new PetResource($pet);
+    }
+
+    /**
+     * Store the pet.
+     * @param  Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1142)